photon-ml/src/main/scala/com/linkedin/photon/ml/function/svm/DistributedSmoothedHingeLossFunction.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.function.svm
+
+import breeze.linalg.Vector
+import org.apache.spark.SparkContext
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.rdd.RDD
+
+import com.linkedin.photon.ml.data.LabeledPoint
+import com.linkedin.photon.ml.function.{DiffFunction, DistributedObjectiveFunction, L2RegularizationDiff}
+import com.linkedin.photon.ml.normalization.NormalizationContext
+import com.linkedin.photon.ml.optimization.{GLMOptimizationConfiguration, RegularizationType}
+import com.linkedin.photon.ml.util.Utils
+
+/**
+ * This class is used to calculate the value and gradient of Rennie's smoothed hinge loss function, as an approximation
+ * of a linear SVM, for distributed optimization problems.
+ *
+ * FAQ: Why use cumGradient (cumulative gradient)?
+ * A:   Using cumGradient allows the functions to avoid memory allocation by modifying and returning cumGradient instead
+ *      of creating a new gradient vector.
+ *
+ * @param sparkContext The Spark context
+ * @param treeAggregateDepth The depth used by treeAggregate. Depth 1 indicates normal linear aggregate. Using
+ *                           depth > 1 can reduce memory consumption in the Driver and may also speed up the
+ *                           aggregation. It is experimental currently because treeAggregate is unstable in Spark
+ *                           versions 1.4 and 1.5.
+ */
+protected[ml] class DistributedSmoothedHingeLossFunction(sparkContext: SparkContext, treeAggregateDepth: Int)
+  extends DistributedObjectiveFunction(sparkContext, treeAggregateDepth)
+  with DiffFunction {
+
+  /**
+   * Compute the value of the function over the given data for the given model coefficients.
+   *
+   * @param input The given data over which to compute the objective value
+   * @param coefficients The model coefficients used to compute the function's value
+   * @param normalizationContext The normalization context
+   * @return The computed value of the function
+   */
+  override protected[ml] def value(
+    input: RDD[LabeledPoint],
+    coefficients: Broadcast[Vector[Double]],
+    normalizationContext: Broadcast[NormalizationContext]): Double =
```

Comment:
Minor: the indentation here is off: arguments should be indented 4 spaces rather than 2. Check other functions in the PR also, there are a few other cases.